solidity/contracts/relay/RelayMaintainerProxy.sol

```diff
@@ -0,0 +1,100 @@
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@keep-network/random-beacon/contracts/Reimbursable.sol";
+import "@keep-network/random-beacon/contracts/ReimbursementPool.sol";
+
+import "./LightRelay.sol";
+
+/// @title LightRelayMaintainerProxy
+/// @notice The proxy contract that allows the relay maintainers to be refunded
+///         for the spent gas from the `ReimbursementPool`. When proving the
+///         next Bitcoin difficulty epoch, the maintainer calls the
+///         `LightRelayMaintainerProxy` which in turn calls the actual `LightRelay`
+///         contract.
+contract LightRelayMaintainerProxy is Ownable, Reimbursable {
+    ILightRelay public lightRelay;
+
+    /// @notice Stores addresses that can maintain the relay.
+    mapping(address => bool) public relayMaintainers;
+
+    /// @notice Gas that is meant to balance the retarget overall cost. Can be
+    //          updated by the governance based on the current market conditions.
+    uint256 public retargetGasOffset;
+
+    event RelayUpdated(address newRelay);
+
+    event RelayMaintainerStatusChanged(
+        address indexed maintainer,
+        bool isAuthorized
+    );
+
+    modifier onlyRelayMaintainer() {
+        require(relayMaintainers[msg.sender], "Caller is not authorized");
+        _;
+    }
+
+    constructor(ILightRelay _lightRelay, ReimbursementPool _reimbursementPool) {
+        lightRelay = _lightRelay;
+        reimbursementPool = _reimbursementPool;
+
+        // TODO: Set the proper value or remove completely.
+        retargetGasOffset = 0;
+    }
+
+    /// @notice Wraps `LightRelay.retarget` call and reimburses the caller's
+    ///         transaction cost. Can only be called by an authorized relay
+    ///         maintainer.
+    /// @dev See `LightRelay.retarget` function documentation.
+    function retarget(bytes memory headers) external onlyRelayMaintainer {
+        uint256 gasStart = gasleft();
+
+        lightRelay.retarget(headers);
+
+        reimbursementPool.refund(
+            (gasStart - gasleft()) + retargetGasOffset,
+            msg.sender
+        );
+    }
+
+    /// @notice Allows the governance to upgrade the `LightRelay` address.
+    /// @dev The function does not implement any governance delay and does not
+    ///      check the status of the `LightRelay`. The Governance implementation
+    ///      needs to ensure all requirements for the upgrade are satisfied
+    ///      before executing this function.
+    function updateLightRelay(ILightRelay _lightRelay) external onlyOwner {
+        lightRelay = _lightRelay;
+
+        emit RelayUpdated(address(_lightRelay));
+    }
+
+    /// @notice Sets authorization status from the given address. Can only be
+    ///         called by the owner.
+    /// @param maintainer The address of the maintainer to be authorized or
```

Comment:
Should we add a similar `@dev` note about the lack of governance delay as in `updateLightRelay`?

Reply:
Done in 902c2411346dfc8b79c8a47f91674bf33d5f89ee.
(Note that I replaced a single function for setting status of a maintainer with `authorize` and `deauthorize` functions).